librecad/src/lib/engine/rs_ellipse.cpp

```diff
@@ -1372,6 +1372,36 @@ void RS_Ellipse::scale(const RS_Vector& center, const RS_Vector& factor) {
     scaleBorders(center,factor);
 // calculateBorders();
 
+}*/
+
+
+void RS_Ellipse::scale(const RS_Vector& center, const RS_Vector& factor)
+{
+    data.center.scale(center, factor);
+   
+    // Get quadratic equation of current ellipse
+    const LC_Quadratic&& eqn = getQuadratic();
+
+    boost::numeric::ublas::matrix<double> eqn_Quad = eqn.getQuad();
+    double const& eqn_a = eqn_Quad(0,0);
+    double const& eqn_c=2.* eqn_Quad(0,1);
+    double const& eqn_b = eqn_Quad(1,1);
+    
+    // quadratic equation of scaled ellipse
+    LC_Quadratic scaled_eqn = eqn;
+ 
+    scaled_eqn.m_mQuad(0,0) = eqn_a * factor.y * factor.y;
```

Comment:
the scaling is correct. Please carry out test to show the old and new scale() methods generate the same results for several test cases. Please design several test cases for unit testing.